### PR TITLE
Contribution guidelines

### PR DESCRIPTION
**Pull request by thesmithee**
Adding the contribution guidelines for .NET Core Conceptual docs. These 
will also be reused in the CoreFX and CoreCLR repos.

/cc @richlander @terrajobst @ellismg

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+updates to your branch. When they are ready to accept the PR, they will add a
+"LGTM" (Looks Good To Me) comment and any other steps that are (maybe) needed.
+
+## Common Pitfalls ##
+
+Below are some common pitfalls you should try to avoid:
+
+- Don't forget to submit an issue before starting work on an article
+- Don't forget to create a separate branch before working on your article
+- Don't update or `merge` your branch after you submit your pull request
```
Comment (richlander) [inline CONTRIBUTING.md]: I don't know what this specific piece of guidance means/is for.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+**Step 5:** Submit a Pull Request from your branch to `dotnet/core-docs/master`.
+
+**Step 6:** Discuss the Pull Request with the .NET team; make any requested
+updates to your branch. When they are ready to accept the PR, they will add a
+"LGTM" (Looks Good To Me) comment and any other steps that are (maybe) needed.
+
+## Common Pitfalls ##
+
+Below are some common pitfalls you should try to avoid:
```
Comment (richlander) [inline CONTRIBUTING.md]: This wording suggests a list of "common pitfalls", but its really a list of guidance.

File: CONTRIBUTING.md (hunk 1 above)
Comment (blackdwarf) [inline CONTRIBUTING.md]: For ASP.NET, since they insist you squash your commits, this is valid, for us not.

**Comment by blackdwarf** [comment]
OK, I'm going to merge this PR, since it is now very old and I think we need these.

**Comment by terrajobst** [comment]
That's OK but any reason the feedback wasn't addressed?

File: CONTRIBUTING.md (hunk 2 above)
Comment (terrajobst) [inline CONTRIBUTING.md]: +1

File: CONTRIBUTING.md (hunk 1 above)
Comment (terrajobst) [inline CONTRIBUTING.md]: So should we remove that line then?

Comment (terrajobst) [inline CONTRIBUTING.md]: So should we remove that line then?

**Comment by blackdwarf** [comment]
I thought I did. Err, lemme check the comments you made, I was pretty sure I corrected for all of them.

**Pull request merged**